### ISSUE TITLE
choco's .NET version is EOL

### ISSUE THREAD

**Issue by stephengtuggy** [chocolatey/choco#1008]
### What You Are Seeing?

When I started to install Chocolatey on Windows 7 Professional SP1 64-bit, it tried to install .NET Framework 4.0. This version of .NET is no longer supported by Microsoft, nor does it receive security updates any longer, AFAIK. https://support.microsoft.com/en-us/lifecycle/search?alpha=.net%20framework

### What is Expected?

Chocolatey should use a supported .NET version, such as 4.5.2, 4.6, 4.6.1, or .NET Core 1.x. (That last one might be a lot more work, though.)

### How Did You Get This To Happen? (Steps to Reproduce)

Perform a fresh install of Windows 7 Professional SP1 64-bit in a VM running under Oracle VirtualBox 5.1.6, with Ubuntu 16.04.1 LTS x64 as the host OS. Make sure the VM has at least 4GB of RAM allotted to it. Set Windows Update to check for updates, but ask you before downloading or installing them. Then check for updates.

In my experience, just checking for updates the first time will use ALL 4 gigs of RAM, and it will take over 36 hours. (!!!)

Once it finishes checking for updates, install all the critical updates offered, except for IE11. If you try to install IE11 along with everything else, it may not work. Instead, install IE11 separately, after rebooting Windows. Then reboot again.

Finally, begin installing Chocolatey using installation method #2 from chocolatey.org (the PowerShell 2.0 method).

### Output Log

https://gist.github.com/stephengtuggy/bfd7f724b08bb74e0c20bb33c1e74d9a

**Comment by ferventcoder** [comment]
@stephengtuggy as mentioned previously, .NET Framework 4.x is the ***minimum*** version that is necessary. All 4.x versions are drop in upgrades that Chocolatey works fine with.

**Comment by ferventcoder** [comment]
We are moving towards .NET Core so we can support Nano. It's on our roadmap - https://chocolatey.org/docs/roadmap

![image](https://cloud.githubusercontent.com/assets/63502/19271655/2abbecf2-8f8b-11e6-93d7-8dcb2ca73107.png)

**Comment by ferventcoder** [comment]
Note that Windows Server 2012R2 comes with .NET 4.5 and Chocolatey installs and runs just fine there. 👍

**Comment by ferventcoder** [comment]
The fact that I don't see this message in your logs tells me you are running on at least .NET 4.5:

~~~sh
!!WARNING!!
Choco prefers to use TLS v1.2 if it is available, but this client is 
 running on .NET 4.0, which uses an older SSL. It's using TLS 1.0 or 
 earlier, which makes it susceptible to BEAST and also doesn't 
 implement the 1/n-1 record splitting mitigation for Cipher-Block 
 Chaining. Upgrade to at least .NET 4.5 at your earliest convenience.
 For more information you should visit https://www.howsmyssl.com/
~~~

**Comment by ferventcoder** [comment]
> Chocolatey should use a supported .NET version, such as 4.5.2, 4.6, 4.6.1, or .NET Core 1.x. (That last one might be a lot more work, though.)

I'm going to close this as already implemented, because it already supports 4.5.x and 4.6.x. Please let me know if you have any further questions. Thanks!

**Comment by stephengtuggy** [comment]
Yes, I am running on .NET Framework 4.6.1. That's because I canceled out of the Chocolatey installation when it tried to install .NET 4.0. Then I manually installed .NET Framework 4.6.1; installed Chocolatey again; and everything was fine.

So the log is rather misleading, I'm afraid. I just included it because that appeared to be one of the required components of a Chocolatey bug report.

I fully realize that Chocolatey **supports** .NET 4.6.x, etc. Really, the point of this issue was that on Win 7, the choco installer installs an old version of .NET by default.

My bug title wasn't very helpful in conveying that. I apologize. I think I will either change it, or open a new issue, with a clearer title.
